packages/react/src/time-input/docs/overview.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: Time input
+description: The Time input component allows users to enter a time free-form, and in various formats.
+group: Forms
+storybookPath: /story/Forms-TimeInput--basic
+unreleased: true
+figmaGalleryNodeId: REPLACE_ME
+relatedComponents: ['text-input']
+---
+
+```jsx live
+() => {
+	const [value, setValue] = React.useState({ raw: '' });
+
+	const onChange = (timeValue) => {
+		console.log('onChange', timeValue);
+		setValue(timeValue);
+	};
+
+	const invalid = Boolean(value.raw) && !isValidTime(value.raw);
+
+	return (
+		<TimeInput
+			invalid={invalid}
```

Comment:
It's a controlled component and has no concept of whether it itself is invalid. I can remove the code, but I thought it be useable is more useful

Reply:
The component knows what the value of `value` is. The component can easily run the invalid date check internally instead of the dev having to write out this same check every time the component is used.

Reply:
But it's a controlled component, just like date picker etc. All of our inputs require consumers to pass when a thing is invalid

Reply:
Why force consumers to write this code out 100 times when it can easily be baked directly into the component?

```js
const invalid = value && !isValidTime(value.raw);
```

Reply:
We can still provide the ability to provide custom `invalid` validation. Checking if the value is a valid time should be hardcoded directly into the component though. It makes no sense to have the easiest implementation method to default to not throwing an error when the time value is invalid.

Reply:
@stowball ok I see what your point is now. I was thinking from the perspective of instant validation. I wasn't aware that it is a requirement to only validate on form submission.

CC @ChrisLaneAU